compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -565,124 +551,237 @@ createProjectPackageDb opts fps = do
-                                  guard $ all (`elem` '.':['0' .. '9']) ver
-                                  Just (uId, Just ver)
-                      let (cfPath, cfBs) =
-                              mkConfFile
-                                  PackageConfigFields
-                                      { pName = pkgName
-                                      , pSrc = "" -- not used
-                                      , pExposedModules = Nothing
-                                      , pVersion = pkgVersion
-                                      , pDependencies = deps
-                                      , pSdkVersion = "unknown"
-                                      , cliOpts = Nothing
-                                      }
-                                  (map T.unpack $ LF.packageModuleNames dalf)
-                                  pkgIdStr
-                      B.writeFile (dbPathAbs </> "package.conf.d" </> cfPath) cfBs
-                      callProcess
-                          (ghcPkgPath </> exe "ghc-pkg")
-                          [ "recache"
-                          -- ghc-pkg insists on using a global package db and will try
-                          -- to find one automatically if we don’t specify it here.
-                          , "--global-package-db=" ++ (dbPathAbs </> "package.conf.d")
-                          , "--expand-pkgroot"
-                          ]
-       | length uniqSdkVersions <= 1 ->
-           do forM_ dalfs $ \dalf -> do
-                  let path = dbPath </> ZipArchive.eRelativePath dalf
-                  createDirectoryIfMissing True (takeDirectory path)
-                  BSL.writeFile path (ZipArchive.fromEntry dalf)
-              forM_ confFiles $ \conf ->
-                  BSL.writeFile
-                      (dbPath </> "package.conf.d" </>
-                       (takeFileName $ ZipArchive.eRelativePath conf))
-                      (ZipArchive.fromEntry conf)
-              forM_ srcs $ \src -> do
-                  let path = dbPath </> ZipArchive.eRelativePath src
-                  write path (ZipArchive.fromEntry src)
-              ghcPkgPath <- getGhcPkgPath
-              callProcess
-                  (ghcPkgPath </> exe "ghc-pkg")
-                  [ "recache"
-                  -- ghc-pkg insists on using a global package db and will try
-                  -- to find one automatically if we don’t specify it here.
-                  , "--global-package-db=" ++ (dbPath </> "package.conf.d")
-                  , "--expand-pkgroot"
-                  ]
+                              dbPathAbs
+                              projectPackageDatabaseAbs
+                              unitIdStr
+                              pkgIdStr
+                              pkgName
+                              mbPkgVersion
+                              deps
+
+       | length uniqSdkVersions <= 1 -> installDar dbPath edConfFiles edDalfs edSrcs
```

Comment:
If we have dalf imports we always do 3). I also think the different use cases are fairly confusing, in particular since this a DAML only thing. What fields with what semantics would you suggest in `daml.yaml`?

Reply:
we could have a field `imports` that allows only dars of the same sdk,  and one `data-imports` where we allow dalfs and different sdk versions.

Reply:
Something like that but let’s keep that for a separate PR. I’ll take another quick look tomorrow morning before approving but I think this PR should be good to land now :+1: